tests/bsim/bluetooth/mesh/src/test_cdp1.c

```diff
@@ -0,0 +1,517 @@
+	}
+};
+
+static const struct bt_mesh_comp_p1_model_item p1_cfg_srv_mod = {
+	.cor_present = 0,
+	.format = 0,
+	.ext_item_cnt = 0,
+};
+
+static const struct bt_mesh_comp_p1_model_item p1_cfg_cli_mod = {
+	.cor_present = 0,
+	.format = 0,
+	.ext_item_cnt = 0,
+};
+
+static const struct bt_mesh_comp_p1_model_item p1_mod1 = {
+	.cor_present = 0,
+	.format = 0,
+	.ext_item_cnt = 0,
+};
+
+static const struct bt_mesh_comp_p1_model_item p1_mod2 = {
+	.cor_present = 0,
+	.format = 0,
+	.ext_item_cnt = 1,
+};
+
+static const struct bt_mesh_comp_p1_model_item p1_mod3 = {
+	.cor_present = 1,
+	.format = 0,
+	.ext_item_cnt = 1,
+	.cor_id = 0,
+};
+
+static const struct bt_mesh_comp_p1_model_item p1_mod4 = {
+	.cor_present = 0,
+	.format = 0,
+	.ext_item_cnt = 0,
+};
+
+static const struct bt_mesh_comp_p1_model_item p1_mod5 = {
+	.cor_present = 0,
+	.format = 0,
+	.ext_item_cnt = 0,
+};
+
+static const struct bt_mesh_comp_p1_model_item p1_mod6 = {
+	.cor_present = 0,
+	.format = 1,
+	.ext_item_cnt = 1,
+};
+
+static const struct bt_mesh_comp_p1_model_item p1_vnd1 = {
+	.cor_present = 1,
+	.format = 0,
+	.ext_item_cnt = 0,
+	.cor_id = 0,
+};
+
+static const struct bt_mesh_comp_p1_model_item p1_elem0_models[] = {
+	p1_cfg_srv_mod,
+	p1_cfg_cli_mod,
+	p1_mod1,
+	p1_mod2,
+	p1_mod3,
+	p1_vnd1,
+};
+
+static const struct bt_mesh_comp_p1_model_item p1_elem1_models[] = {
+	p1_mod4,
+};
+
+static const struct bt_mesh_comp_p1_model_item p1_elem2_models[] = {
+	p1_mod5,
+};
+
+static const struct bt_mesh_comp_p1_model_item p1_elem3_models[] = {
+	p1_mod5,
+};
+
+static const struct bt_mesh_comp_p1_model_item p1_elem4_models[] = {
+	p1_mod6,
+};
+
+static const struct bt_mesh_comp_p1_model_item *p1_elem_models[] = {
+	p1_elem0_models,
+	p1_elem1_models,
+	p1_elem2_models,
+	p1_elem3_models,
+	p1_elem4_models,
+};
+
+static const struct bt_mesh_comp_p1_elem p1_elem0 = {
+	.nsig = 5,
+	.nvnd = 1,
+};
+
+static const struct bt_mesh_comp_p1_elem p1_elem1 = {
+	.nsig = 1,
+	.nvnd = 0,
+};
+
+static const struct bt_mesh_comp_p1_elem p1_elem2 = {
+	.nsig = 1,
+	.nvnd = 0,
+};
+
+static const struct bt_mesh_comp_p1_elem p1_elem3 = {
+	.nsig = 1,
+	.nvnd = 0,
+};
+
+static const struct bt_mesh_comp_p1_elem p1_elem4 = {
+	.nsig = 1,
+	.nvnd = 0,
+};
+
+static struct bt_mesh_comp_p1_elem p1_elems[] = {
+	p1_elem0,
+	p1_elem1,
+	p1_elem2,
+	p1_elem3,
+	p1_elem4,
+};
+
+static void provision_and_configure(struct bt_mesh_test_cfg cfg)
+{
+	int err;
+	uint8_t status;
+
+	err = bt_mesh_provision(net_key, 0, 0, 0, cfg.addr, cfg.dev_key);
+	if (err) {
+		FAIL("Provisioning failed (err %d)", err);
+		return;
+	}
+
+	err = bt_mesh_cfg_cli_app_key_add(0, cfg.addr, 0, 0, app_key, &status);
+	if (err || status) {
+		FAIL("AppKey add failed (err %d, status %u)", err, status);
+		return;
+	}
+}
+
+static void verify_cdp1(struct bt_mesh_comp_p1_elem p1_elem,
+			struct bt_mesh_comp_p1_model_item mod_item,
+			struct bt_mesh_comp_p1_ext_item ext_item, struct net_buf_simple p1_dev_comp)
```

Comment:
It is too much to pass these structures by values:
```suggestion
static void verify_cdp1(struct bt_mesh_comp_p1_elem *p1_elem,
			struct bt_mesh_comp_p1_model_item *mod_item,
			struct bt_mesh_comp_p1_ext_item *ext_item, struct net_buf_simple *p1_dev_comp)
```

Reply:
Done!